pkg/util/azure/credential.go

```diff
@@ -129,5 +129,14 @@ func newConfigCredential(creds map[string]string, options configCredentialOption
 		return nil, errors.Errorf("%s is required", CredentialKeyPassword)
 	}
 
+	//msiEndpoint
+	msiEndpoint := creds[CredentialMSI_ENDPOINT]
```

Comment:
I don't think we should need to set this.
this is a well known address in the library generally

similarly ideneity endpoint.

Reply:
Thanks for quick review.
So ```	CredentialIMDS_ENDPOINT              = "IMDS_ENDPOINT"              // #nosec
	CredentialIDENTITY_ENDPOINT          = "IDENTITY_ENDPOINT"          // #nosec
	CredentialIDENTITY_HEADER            = "IDENTITY_HEADER"            // #nosec
	CredentialIDENTITY_SERVER_THUMBPRINT = "IDENTITY_SERVER_THUMBPRINT" // #nosec
	CredentialMSI_ENDPOINT               = "MSI_ENDPOINT"               // #nosec
``` also not need to notice the user too? 

I will omit it in next rebase.

Reply:
basically like I called out in issue
MSI based auth should just work with 3 params.
AZURE_CLIENT_ID=
AZURE_TENANT_ID=
AZURE_CLOUD_NAME=

nothing else is needed to be set.

Reply:
thanks , i rebased the code.